ocs_sample_library_preview/SharedStreams.py

```diff
@@ -0,0 +1,483 @@
+
+        response = self.__base_client.request(
+            'get',
+            self.__stream_path.format(
+                tenant_id=tenant_id,
+                namespace_id=namespace_id,
+                stream_id=self.__base_client.encode(stream_id)),
+            headers=headers)
+        self.__base_client.checkResponse(
+            response, f'Failed to get SdsStream, {stream_id}.')
+
+        result = SdsStream.fromJson(response.json())
+        return result
+
+    def getStreamType(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str) -> SdsType:
+        """
+        Retrieves a stream type specified by 'stream_id' in a community specified by 'community_id' from the Sds Service
+        :param tenant_id: tenant to work against
+        :param namespace_id: namespace to work against
+        :param community_id: community to work against
+        :param stream_id: id of the stream
+        :return: the stream type as an SdsType
+        """
+        self.__validateParameters(tenant_id, namespace_id, community_id, stream_id)
+        headers = self.__base_client.getCommunityRequestHeaders(community_id=community_id)
+
+        response = self.__base_client.request(
+            'get',
+            self.__stream_path.format(
+                tenant_id=tenant_id,
+                namespace_id=namespace_id,
+                stream_id=self.__base_client.encode(stream_id)) + '/Type',
+            headers=headers)
+        self.__base_client.checkResponse(
+            response, f'Failed to get SdsStream type, {stream_id}.')
+
+        result = SdsType.fromJson(response.json())
+        return result
+
+    def getStreams(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str, query: str = '', skip: int = 0,
+                   count: int = 100) -> list[SdsStream]:
+        """
+        Retrieves a list of streams associated with 'namespace_id' under the specified tenant and community
+        :param tenant_id: tenant to work against
+        :param namespace_id: namespace to work against
+        :param community_id: community to work against
+        :param stream_id: id of the stream
+        :param query: filtering query
+        :param skip: number of streams to skip for paging
+        :param count: number of streams to limit to
+        :return: array of SdsStreams
+        """
+        self.__validateParameters(tenant_id, namespace_id, community_id, stream_id)
+        headers = self.__base_client.getCommunityRequestHeaders(community_id=community_id)
+
+        response = self.__base_client.request(
+            'get',
+            self.__streams_path.format(
+                tenant_id=tenant_id,
+                namespace_id=namespace_id),
+            params={'query': query, 'skip': skip, 'count': count},
+            headers=headers)
+        self.__base_client.checkResponse(
+            response, 'Failed to get all SdsStreams.')
```

Comment:
I might be misreading this, but I think streamId shouldn't be in this function as a required (and validated) parameter

Reply:
Thank you for catching that, streamId should not be in this method at all, removed.